File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int height(TreeNode* root){
        if(root==NULL){
            return 0;
        }
        return 1+max(height(root->left), height(root->right));
    }

    void printCurrentLevel(vector<int>& ans,TreeNode* root, int level){
        if(root==NULL){
            return;
        }
        if(level==1){
            ans.push_back(root->val);
        }
        else if(level>1){
            printCurrentLevel(ans, root->left, level-1);
            printCurrentLevel(ans, root->right, level-1);
        }
    }
    vector<vector<int>> levelOrder(TreeNode* root) {
        int h = height(root);
        vector<vector<int>> result;
        for(int i=1;i<=h;i++){
            vector<int> ans;
            printCurrentLevel(ans, root, i);
            result.push_back(ans);
        }
        return result;
    }
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int height(TreeNode root){
        if(root==null){
            return 0;
        }
        return 1+Math.max(height(root.left), height(root.right));
    }

    void printCurrentLevel(List<Integer> ans,TreeNode root, int level){
        if(root==null){
            return;
        }
        if(level==1){
            ans.add(root.val);
        }
        else if(level>1){
            printCurrentLevel(ans, root.left, level-1);
            printCurrentLevel(ans, root.right, level-1);
        }
    }

    List<List<Integer>> levelOrder(TreeNode root) {
        int h = height(root);
        List<List<Integer>> result = new ArrayList<>();
        for(int i=1;i<=h;i++){
            List<Integer> ans = new ArrayList<>();
            printCurrentLevel(ans, root, i);
            result.add(ans);
        }
        return result;
    }
}